hash_practice/exercises.py

```diff
@@ -22,8 +65,24 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)

```

Comment:
Assuming `n` is the number of rows/columns, this is correct. Though arguably since we only use 3 for `n`, we can probably just consider this O(1).